### PR TITLE
Demo user

### PR DESCRIPTION
**Pull request by tejasj1718**
### What this PR does (required):
- Adds a demo user link below the login form to access the demo account.
- Adds a route to login as a demo user based on credentials in the .env file.

### Screenshots / Videos (front-end only):
Demo User Button
PRDemoUser

### Any information needed to test this feature (required):
- Add the details of the DEMO user to the .env file.
- Run the server and the client and test the demo user by clicking the link.

### Any issues with the current functionality (optional):
- Design suggestions?
- This branches off from the styled login pages PR. this was necessary as the styling was needed to implement this PR.

### CODE REVIEW

**Comment by ezeilo-su** [comment]
Hey @tejasj1718, seems this PR is the same with [#48](https://github.com/hatchways/team-granite/pull/48)?

**Comment by tejasj1718** [comment]
Hey @ezeilo-su , this PR is based of off a feature mentioned in #48. I did that because the issue says to maintain consistent styling of the demo user button.

File: client/src/pages/SignUp/useStyles.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { makeStyles } from '@material-ui/core/styles';
 
 const useStyles = makeStyles(() => ({
   root: {
-    minHeight: '100vh',
+    // minHeight: '100px',
```
Comment (ezeilo-su) [inline client/src/pages/SignUp/useStyles.ts:5]: Let's remove unused code please

File: server/controllers/auth.js [hunk 2]
```diff
@@ -114,3 +114,45 @@ exports.logoutUser = asyncHandler(async (req, res, next) => {
 
   res.send("You have successfully logged out");
 });
+
+// @route GET /auth/demo
+// @desc Login user
+// @access Public
```
Comment (ezeilo-su) [inline server/controllers/auth.js:120]: Nice documentation

File: server/controllers/auth.js [hunk 3]
```diff
@@ -114,3 +114,45 @@ exports.logoutUser = asyncHandler(async (req, res, next) => {
...
+
+  let user = await User.findOne({ email });
+
+  if (!user) {
+    user = await User.create({
+      email,
+      username,
+      password,
+    });
+  }
```
Comment (ezeilo-su) [inline server/controllers/auth.js:134]: You're supposed to return a `404` if no user exists with this email. A separate route is provided for registering a new user, so shouldn't be done here.

**Review by ezeilo-su — COMMENTED** [review]
Good work with this one :+1:  Please take a look at the comments I left below.

File: server/controllers/auth.js (hunk 2 above)
Comment (tejasj1718) [reply server/controllers/auth.js:120]: Thanks!

File: server/controllers/auth.js (hunk 3 above)
Comment (tejasj1718) [reply server/controllers/auth.js:134]: Hey, the reason behind this is cause I am implementing the demo user feature and the demo user is based of the .env file. So if the demo user exists log him in, if not create one automatically. This allows for a seamless experience.

File: client/src/pages/SignUp/SignUp.tsx [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 import useStyles from './useStyles';
 import register from '../../helpers/APICalls/register';
-import SignUpForm from './SignUpForm/SignUpForm';
-import AuthHeader from '../../components/AuthHeader/AuthHeader';
 import { useAuth } from '../../context/useAuthContext';
 import { useSnackBar } from '../../context/useSnackbarContext';
+import { Link } from 'react-router-dom';
+import { Divider } from '@material-ui/core';
+import SignUpForm from './SignUpForm/SignUpForm';
+import loginDemoUser from '../../helpers/APICalls/loginDemoUser';
```
Comment (moffatethan) [inline client/src/pages/SignUp/SignUp.tsx:12]: Are these duplicated from #48?

Let's adjust the base of the branch to that one so we see relevant changes.

**Review by moffatethan — CHANGES_REQUESTED** [review]

**Comment by moffatethan** [comment]
Let's also fix the merge conflicts too please!

File: client/src/pages/SignUp/SignUp.tsx (hunk 4 above)
Comment (tejasj1718) [reply client/src/pages/SignUp/SignUp.tsx]: done!

**Review by moffatethan — APPROVED** [review]

**Pull request merged**